feat:  Titel explore ,for you and animated category tabs

Mohammed-qmr opened the pull request:
improvements to the designSystem :

Section titles: Added "For You" and "Explore" headers to clearly separate content areas.

Animated category tabs: Implemented smooth transitions between category tabs for better user experience.

Mohammed-qmr committed:
feat: feat:  section headers, and animated category tabs to Search screen

fatimafaisalghazi commented on designSystem/src/main/java/com/example/designsystem/component/CustomTextTitel.kt:
```diff
@@ -0,0 +1,66 @@
+    secondaryText: String? = null,
+    endIcon: Painter? = null,
+    onSeeAllClick: (() -> Unit)? = null,
+    modifier: Modifier = Modifier
```
make modifier as the first optional paramrter

Mohammed-qmr replied:
Why  make it into an optional parameter ?

fatimafaisalghazi replied:
it is already an optional paramete 
it mean the it is have a default value

fatimafaisalghazi commented on designSystem/src/main/java/com/example/designsystem/component/CustomTextTitel.kt:
```diff
@@ -0,0 +1,66 @@
+//        onSeeAllClick = { /* Handle click */ }
+//    )
+//}
+
```
try not to push a dead code

fatimafaisalghazi commented on designSystem/src/main/java/com/example/designsystem/component/FilterBar.kt:
```diff
@@ -67,21 +71,63 @@ fun FilterChip(
+//        selectedItem = "All",
+//        onItemClick = {}
+//    )
+//
```
same

fatimafaisalghazi commented on designSystem/src/main/java/com/example/designsystem/component/FilterBar.kt:
```diff
@@ -67,21 +71,63 @@ fun FilterChip(
+                    modifier = Modifier
+                            .padding(top = AppTheme.spacing.extraSmall)
+                        .height(1.dp)
+                        .clip(RoundedCornerShape(16.dp))
```
try to use the radius from the design system

fatimafaisalghazi commented on designSystem/src/main/java/com/example/designsystem/component/FilterBar.kt:
```diff
@@ -67,21 +71,63 @@ fun FilterChip(
+    )
+    val textWidthDp = with(density) { textLayoutResult.size.width.toDp() }
+    val textColor =
+        if (isSelected) AppTheme.colors.surfaceColor.surface else AppTheme.colors.surfaceColor.onSurfaceVariant
```
use animateColorAsState instead of if statement

fatimafaisalghazi commented on designSystem/src/main/java/com/example/designsystem/component/FilterBar.kt:
```diff
@@ -67,21 +71,63 @@ fun FilterChip(
     modifier: Modifier = Modifier
 ) {
-    val backgroundColor = if (isSelected) AppTheme.colors.brandColors.primary else AppTheme.colors.surfaceColor.onSurfaceContainer
-    val textColor = if (isSelected) AppTheme.colors.surfaceColor.onSurface else AppTheme.colors.surfaceColor.onSurface_1
```
use animateColorAsState instead of if statement

Mohammed-qmr committed:
improve CustomBottomBar code clarity

- Moved modifier to be the first optional parameter in composables
- Cleaned up unused code
- Applied animateColorAsState for smoother transitions
- Used shape radius from DesignSystem for consistency

ahmedhgabr commented on designSystem/src/main/java/com/example/designsystem/component/FilterBar.kt:
```diff
@@ -1,53 +1,61 @@
 package com.example.designsystem.component
+
+import androidx.camera.core.Preview
```
This line prevents the build. I believe you just need to remove it.
<img width="803" height="250" alt="image" src="https://github.com/user-attachments/assets/5c77eebe-1a55-4333-b52f-a2377f60c8a2" />

ahmedhgabr commented on designSystem/src/main/java/com/example/designsystem/component/FilterBar.kt:
```diff
@@ -67,21 +73,58 @@ fun FilterChip(
+            }
+        }
+    }
+}@Composable
```
Spacing

ahmedhgabr commented on designSystem/src/main/java/com/example/designsystem/component/FilterBar.kt:
```diff
@@ -1,53 +1,61 @@
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.animation.animateColorAsState
 import androidx.compose.foundation.background
 import androidx.compose.foundation.clickable
```
`ctrl + Alt + o`  removes unused imports. This is silly, but I just wanted to share information.

ahmedhgabr commented on designSystem/src/main/java/com/example/designsystem/component/FilterBar.kt:
```diff
@@ -1,53 +1,61 @@
-
-
 @Composable
 fun FilterbBar(
```
`FilterBar` instead of `FilterbBar`

ahmedhgabr replied:
Or does the b in the name have a meaning?

ahmedhgabr commented on designSystem/src/main/java/com/example/designsystem/component/FilterBar.kt:
```diff
@@ -67,21 +73,58 @@ fun FilterChip(
-
-    }
+    )
 }
```
It will be useful to add a preview at the end

Mohammed-qmr committed:
Merge remote-tracking branch 'origin/develop' into feature/search-tabs-titles

Mohammed-qmr committed:
Delete .kotlin/errors/errors-1752246634198.log

Alaakhaled101 committed:
Merge branch 'develop' into feature/search-tabs-titles

karrar-abbas commented:
resolve conflict caused by commit files that should not uploaded to github (ide files)
